app/scripts/migrations/054.js

```diff
@@ -0,0 +1,57 @@
+import { cloneDeep } from 'lodash';
+
+const version = 54;
+
+/**
+ * Remove tokens that don't have an address due to
+ * lack of previous addToken validation.  Also removes
+ * an unwanted, undefined image property
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+
+    const { PreferencesController } = versionedData.data;
+
+    // These properties are arrays and can be directly filtered
+    const tokenArrays = ['tokens', 'hiddenTokens'];
+    tokenArrays.forEach((tokenKey) => {
+      if (PreferencesController[tokenKey]) {
+        PreferencesController[tokenKey] = PreferencesController[
+          tokenKey
+        ].filter(({ address }) => address);
+      }
+    });
+
+    // These properties are: (account) -> (network) -> [(token), ....]
+    const tokenObjects = ['accountHiddenTokens', 'accountTokens'];
+    tokenObjects.forEach((tokenObject) => {
+      if (PreferencesController[tokenObject]) {
```

Comment:
This `if` is required because neither `PreferencesController. accountHiddenTokens ` or `PreferencesController. accountTokens ` exists in the migration data to this point.